### ISSUE TITLE
massive storm of presence EDUs when joining a large room

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#3120]
after you do your send_join to a room, all the servers in the room seem to immediately send you presence updates. Since this will often be the first contact you get from those servers, we then end up doing hundreds of server key requests, all of which end up starving out the actual join request - contribuiting to the join itself taking ages.

**Comment by richvdh** [comment]
possibly related to #2514

**Comment by ara4n** [comment]
dup  #3962?

**Comment by richvdh** [comment]
> dup #3962?

I don't think so. #3962 is about presence updates that we don't expect. AFAICT the EDUs referred to in this issue are expected - the problem is just that there are a lot of them, and they all arrive *during* the join dance and tend to block it.
